index.js

```diff
@@ -92,4 +89,76 @@ server.put('/api/cohorts/:id', (req, res) => {
 });
 
 
+//students spi requests -----------------------
+server.get('/api/students', (req, res) => {
+    db('students')
+      .then(students => {
+        res.status(200).json(students);
+      })
+      .catch(err => res.status(500).json(err));
```

Comment:
This `get` route looks nearly identical to the one for cohorts. Why not try to at least make one function for their callbacks? For example:
```javascript
const getAll = tableName => (req, res) => {
  db(tableName)
    .then(data =>  res.status(200).json(data));
    .catch(err => res.status(500).json(err));
};
```
and the usage:
```javascript
server.get('/api/cohorts', getAll('cohorts'));
server.get('/api/students', getAll('students'));
```
Those lines look pretty much identical as well. What if we did this iteratively (especially since it'd make it more reusable)? For example:
```javascript
const tables = [ 'cohorts', 'students' ];
tables.forEach(name => {
  server.get(`/api/${name}`, getAll(name));
});
```
Now, let's add another for your post:
```javascript
const post = tableName => (req, res) => {
  db(tableName)
    .insert(req.body)
    .then(([ id ]) => {
      db(name)
        .where({ id })
        .then(data => {
          res.status(201).json(data);
        })
    })
    .catch(err => {
        err.status(500).json(err);
    });
};

// and do it the iterative way that we did the other!
tables.forEach(name => {
  server.get(`/api/${name}`, getAll(name));
  server.post(`/api/${name}`, post(name));
});
```